### ISSUE TITLE
Question about `mixed_precision='fp16'`  on single  GPU  P100

### ISSUE THREAD

**Issue by YiandLi** [huggingface/accelerate#2413]
### System Info

Hello, I am using accelerate for mixed-precision training on a single GPU.

I am a fresh coder  with this package  and met  a problem:
After setting `mixed_precision='fp16'`, I found the output tensor is still float32 type. Why?

```python
model= AutoModelForSequenceClassification.from_pretrained("/kaggle/input/debertav3base")

accelerator = Accelerator(mixed_precision='fp16', cpu=not torch.cuda.is_available())

model = accelerator.prepare(model)

tknz= AutoTokenizer.from_pretrained("/kaggle/input/debertav3base")

a  = tknz("hello", return_tensors='pt')
for i in a:
    a[i]=a[i].to('cuda')
    
model(**a).logits.dtype // ----> torch.float32


```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Expected behavior

I think it should float16 , is  it?

**Comment by muellerzr** [comment]
Accelerate will convert the outputs to fp32 at the end, but the gradients etc are still in fp16 (this is to help with things like metric calculations)
